### PR TITLE
Text input fields

### PR DESCRIPTION
**Pull request by cfarm**
Updates text input styles for form updates in [Design Manual Issue 346](https://github.com/cfpb/design-manual/issues/346)
## Additions
- `@input-disabled` and `@input-disabled-bg` color variables
- `.disabled` input style
## Changes
- `@input-error` updated hex value to match new standard
## Review
- @sonnakim
- @Scotchester 
## Screenshots

Inputs in the Design Manual
screen shot 2015-09-24 at 4 05 13 pm
## Notes
- related default input styles changed in [cf-core PR 46](https://github.com/cfpb/cf-core/pull/46)
- See related change to color variables in [Capital Framework PR 184](https://github.com/cfpb/capital-framework/pull/184) and [generator-cf PR 95](https://github.com/cfpb/generator-cf/pull/95)
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [ ] Passes all existing automated tests
- [ ] New functions include new tests
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged
- [x] Visually tested in supported browsers and devices 
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/cf-forms.less [hunk 1]
```diff
@@ -137,87 +119,14 @@ textarea {
...
-    &.warning {
-        border: 1px solid @input-warning;
-        outline: 1px solid @input-warning;
-    }
-    &.success {
-        border: 1px solid @input-success;
-        outline: 1px solid @input-success;
+    &.disabled {
+        outline: 1px solid @input-disabled;
+        background-color: @input-disabled-bg;
```
Comment (Scotchester) [inline src/cf-forms.less]: Where is `@input-disabled-bg` being defined?

**Comment by Scotchester** [comment]
Where will the field-level error message styling be implemented?

**Comment by cfarm** [comment]
I restored the icon styles, since I'll be using the error icon in the updated validation styles (in upcoming PR)

**Comment by sonnakim** [comment]
@Scotchester @cfarm as we (separately) discussed -
- We're keeping field-level warning and field-level success styles in the code base, even though they're not documented in this latest iteration of the Design Manual. (There are a few live pages that currently use both styles and we don't want them to break.)
- We _may_ add field-level success styles back to this latest DM iteration as there may be a good use case for them. But we will decided on that at a later point. 
- We're going to keep field-level warning styles out of the Design Manual  

Also, page-level warning styles are documented in the Alerts and Error Messaging DM page, not in the Form Fields page.

**Comment by jimmynotjim** [comment]
Is this correct? 
screen shot 2015-09-25 at 1 51 47 pm

That disabled input looks like it's still a 2px border

File: src/cf-forms.less [hunk 2]
```diff
@@ -137,17 +122,12 @@ textarea {
...
         outline: 1px solid @input-error;
     }
-    &.warning {
-        border: 1px solid @input-warning;
-        outline: 1px solid @input-warning;
-    }
-    &.success {
-        border: 1px solid @input-success;
-        outline: 1px solid @input-success;
+    &.disabled {
```
Comment (Scotchester) [inline src/cf-forms.less:157]: Missing closing brace for `&.success` declaration.

File: src/cf-forms.less [hunk 3]
```diff
@@ -137,17 +122,12 @@ textarea {
...
+        border-width: 2px;
+        background-color: @input-disabled;
     }
 }
 
-
```
Comment (Scotchester) [inline src/cf-forms.less:150]: Please leave two blank lines between each Topdoc block.

File: src/cf-forms.less [hunk 4]
```diff
@@ -216,7 +196,6 @@ textarea {
     color: @input-success;
 }
 
-
```
Comment (Scotchester) [inline src/cf-forms.less:219]: Please leave two blank lines between each Topdoc block.

**Comment by cfarm** [comment]
### Updates
- Restored icon styles
- Restored .warning and .success field level styles
- Updated error color to non-brand variable since we have the brand styling in cf-theme-overrides.less
- Fixed spacing

**Comment by cfarm** [comment]
Updated .disabled input border to be 1px.

File: src/cf-forms.less [hunk 5]
```diff
@@ -452,4 +461,4 @@ textarea {
 /* topdoc
   name: EOF
   eof: true
-*/
+*/
```
Comment (Scotchester) [inline src/cf-forms.less]: Please restore EOF blank line

File: src/cf-forms.less [hunk 6]
```diff
@@ -100,9 +104,6 @@
     - name: Error state
       codenotes:
         - .error
-      notes:
-        - "See the 'Form icons' section below for guidance on adding icons to
-           states."
```
Comment (Scotchester) [inline src/cf-forms.less:105]: This comment should be restored, yes?

**Comment by jimmynotjim** [comment]
:+1: after addressing Scott's last two minor comments.

**Comment by cfarm** [comment]
moar fixes added.

**Comment by Scotchester** [comment]
Kind of amusing that all this PR amounted to in the end was adding the disabled state.

**Comment by cfarm** [comment]
image

**Pull request merged**